### ISSUE TITLE
new theme `inherits` directive does not allow for *unsetting* previously-defined options

### ISSUE THREAD

**Issue by nestor-custodio** [helix-editor/helix#5053]
The new theming `inherits` directive is great, but does not allow for certain use cases. If I want to inherit from "monokai_pro_spectrum", which defines ...
```ini
"ui.cursor.primary" = { fg = "base7", modifiers = ["reversed"] }
```
... but I want to set my own cursor style for different modes via ...
```ini
# in "custom.toml"

inherits "monokai_pro_spectrum"

"ui.cursor"        = { fg = "base8x0c", bg = "green" , modifiers = ["bold"] }
"ui.cursor.insert" = { fg = "base8x0c", bg = "red"   , modifiers = ["bold"] }
"ui.cursor.select" = { fg = "base8x0c", bg = "yellow", modifiers = ["bold"] }
```
... this does not produce the intended result, as the `"ui.cursor.primary"` already defined means I don't see the cursor colors I want (unless I create multiple cursors, in which case the primary will be one-off'ed from the others).

Having no way to "unset" the already-defined value (possibly via something like `"ui.cursor.primary" = { }`) greatly limits the usability of the (otherwise phenomenal) new `inherits` feature.

**Comment by 2dav** [comment]
Same issue

```toml
inherits = "gruvbox_dark_hard"

"comment.block" = "orange1"
"comment.line" = "orange1"
"comment.block.documentation" = "aqua1"
```
those `comment` scopes are just ignored like as if parent theme `comment` scope have a higher priority.

**Comment by tysteiman** [comment]
@nestor-custodio I may not be understanding the problem fully, however I modified some of the theme for tokyonight_storm which inherits tokyonight and I was able to clear some of the values using ui.cursor.primary = {}

```
inherits = "tokyonight"

"ui.background" = { }
"ui.linenr" = { }
"ui.cursor.primary" = { }
```

This cleared out those respective styles when I did that.

This is a tmux session with the test build on the left and the normal release on the right.

![image](https://user-images.githubusercontent.com/11522852/206352658-a22350ff-1bd4-49ab-81b9-3838d8078a08.png)

I apologize in advance if I misunderstood you.

**Comment by nestor-custodio** [comment]
@tysteiman There's a subtle difference here in that saying `"ui.cursor.primary" = { }` doesn't mean "pretend we never said anything about ui.cursor.primary" -- it means "I would like the primary cursor to not be visible".

So far as I've been able to determine, the `"ui.cursor.primary"` style takes precedence over `"ui.cursor"`/`"ui.cursor.select"`/`"ui.cursor.insert"`, so once it has been defined, the ability to set the (primary) cursor style for different modes is lost.

To be clear: I'm not saying there's a problem with the fact that the primary cursor setting overrides the mode-specific ones. All I'm saying is that there needs to be some way to say "pretend this thing was never defined" -- either via (a) making `{ }` assignment mean something other than "not visible", or (b) some other special directive (like `inherits`) that lets me provide a list of things I wish the "parent" theme hadn't defined.

**Comment by pascalkuthe** [comment]
I believe that generally you should never need to "unset" a theme keysm, normally you should just be able to set the more specific key to the fallback value you want to use. 

That's not possible in this specific case due to the unusal behaviour of `ui.cursor`. This is a known issue with multiple issues/PRs about it. The latest attempt at solving this is #4596.
You can read trough the discussion linked in the PR if you are interested.

There are multiple issues/PRs about the `ui.cursor.primary` issue already so unless we really such an "uninherit" feature here we could close this issue

**Comment by nestor-custodio** [comment]
Fair enough. Apologies for opening up a new issue where several already existed. Assuming this was a new problem, I only searched for "inherits" when looking for existing issues.

**Comment by pascalkuthe** [comment]
fixed by #5130
